server/modules/protocol/MySQL/mysql_common.cc

```diff
@@ -1768,3 +1767,134 @@ void mxs_mysql_execute_kill_user(MXS_SESSION* issuer, const char* user, kill_typ
+    ptr += 2; // status      
+    ptr += 2; // number of warnings
+
+    if (server_capabilities & GW_MYSQL_CAPABILITIES_SESSION_TRACK)
+    {
+        if (ptr < (local_buf + len)) 
+        {
+            ptr += mxs_leint_consume(&ptr);  // info
+            if (server_status  & SERVER_SESSION_STATE_CHANGED)
+            {
+                mxs_leint_consume(&ptr);    // total SERVER_SESSION_STATE_CHANGED length
+                while (ptr < (local_buf + len))
+                {
+                    enum_session_state_type type = (enum enum_session_state_type)mxs_leint_consume(&ptr);
+                    switch (type)
+                    {
+                        case SESSION_TRACK_STATE_CHANGE:
+                        case SESSION_TRACK_SCHEMA:
+                        case SESSION_TRACK_GTIDS:
+                            ptr += mxs_leint_consume(&ptr);
+                            break;
+                        case SESSION_TRACK_TRANSACTION_CHARACTERISTICS:
+                            mxs_leint_consume(&ptr); //length
+                            var_value = mxs_lestr_consume_dup(&ptr);
+                            gwbuf_add_property(buff, (char *)"trx_characteristics", var_value);
+                            MXS_FREE(var_value);
+                            break;
+                        case SESSION_TRACK_SYSTEM_VARIABLES:
+                            mxs_leint_consume(&ptr); //lenth 
+                            // system variables like autocommit, schema, charset ... 
+                            var_name = mxs_lestr_consume_dup(&ptr);
+                            var_value = mxs_lestr_consume_dup(&ptr);
+                            gwbuf_add_property(buff, var_name, var_value);
+                            MXS_DEBUG("SESSION_TRACK_SYSTEM_VARIABLES, name:%s, value:%s", var_name, var_value);
+                            MXS_FREE(var_name);
+                            MXS_FREE(var_value);
+                            break;
+                        case SESSION_TRACK_TRANSACTION_TYPE:
+                            mxs_leint_consume(&ptr); // length
+                            trx_info = mxs_lestr_consume_dup(&ptr);
+                            MXS_DEBUG("get trx_info:%s", trx_info);
+                            gwbuf_add_property(buff, (char *)"trx_state", trx_info);
+                            MXS_FREE(trx_info);
+                            break;
+                        default:
+                            ptr += mxs_leint_consume(&ptr);
+                            MXS_WARNING("recieved unexpecting session track type:%d", type);
+                            break;
+                    }
+                }
+            }
+        }
+    }
+}
+
+/***
+ * As described in https://dev.mysql.com/worklog/task/?id=6631
+ * When session transation state changed
+ * SESSION_TRACK_TRANSACTION_TYPE (or SESSION_TRACK_TRANSACTION_STATE in MySQL) will
+ * return an 8 bytes string to indicate the transaction state details
```

Comment:
I believe it would be OK to refer to that document and add the following excerpt from it:
```
The following states may be signaled:

Place 1: Transaction.
  T  explicitly started transaction ongoing
  I  implicitly started transaction (@autocommit=0) ongoing
  _  no active transaction

Place 2: unsafe read
  r  one/several non-transactional tables were read
     in the context of the current transaction
  _  no non-transactional tables were read within
     the current transaction so far

Place 3: transactional read
  R  one/several transactional tables were read
  _  no transactional tables were read yet

Place 4: unsafe write
  w  one/several non-transactional tables were written
  _  no non-transactional tables were written yet

Place 5: transactional write
  W  one/several transactional tables were written to
  _  no transactional tables were written to yet

Place 6: unsafe statements
  s  one/several unsafe statements (such as UUID())
     were used.
  _  no such statements were used yet.

Place 7: result-set
  S  a result set was sent to the client
  _  statement had no result-set

Place 8: LOCKed TABLES
  L  tables were explicitly locked using LOCK TABLES
  _  LOCK TABLES is not active in this session
```

Reply:
done